### PR TITLE
MeanIoU: Support y_pred passed as a logits tensor or probability tensor

### LINKED ISSUE #45598: Please let tf.keras.metrics.MeanIoU support y_pred from logits
<em>Please make sure that this is a feature request. As per our [GitHub Policy](https://github.com/tensorflow/tensorflow/blob/master/ISSUES.md), we only address code/doc bugs, performance issues, feature requests and build/installation issues on GitHub. tag:feature_template</em>

**System information**
- TensorFlow version (you are using): 2.4.0
- Are you willing to contribute it (Yes/No): Yes

**Describe the feature and the current behavior/state.**
 tf.keras.metrics.MeanIoU doesn't have an option that allow y_pred passed as logits, we want to pass y_pred as a logits tensor, you can simply add an optional step into the ```update_state``` function of this class: 
```
    y_pred = math_ops.argmax(y_pred, axis=-1)
```
**Will this change the current api? How?**
Yes, add one parameter from_logits, you can keep default as False though:
```tf.keras.metrics.MeanIoU(num_classes, from_logits=False, name=None, dtype=None)```

**Who will benefit with this feature?**
Computer vision object detection/segmentation community

**Any Other info.**

### PR DESCRIPTION
**Pull request by motionlife**
Just like categorical_accuracy and sparse_categorical_accuracy, this change allow MeanIoU metric support y_pred to be passed as a logits tensor.

### CODE REVIEW

**Comment by motionlife** [comment]
@googlebot I signed it!

**Review by bhack — CHANGES_REQUESTED** [review]
You need to add tests for the logit case

**Comment by bhack** [comment]
Fixes https://github.com/tensorflow/tensorflow/issues/45598

**Comment by motionlife** [comment]
> You need to add tests for the logit case

Thanks for reminding, just added :)

**Comment by TylerADavis** [comment]
I've created a PR adding support for passing probabilities to `MeanIoU`, #47410. Could you take a look and let me know if you have any recommendations for the interface? I ask because I see this PR adds a `from_logits=False`, and I'm not sure if the best solution is to have both `from_logits` and `from_probability` as toggles, or if there is a better option.

While this PR also adds support for probabilities, #47410 is different because it does not use argmax, allowing for multilabel classification, as discussed in https://github.com/tensorflow/tensorflow/issues/39173#issuecomment-655022504

**Comment by motionlife** [comment]
> I've created a PR adding support for passing probabilities to `MeanIoU`, #47410. Could you take a look and let me know if you have any recommendations for the interface? I ask because I see this PR adds a `from_logits=False`, and I'm not sure if the best solution is to have both `from_logits` and `from_probability` as toggles, or if there is a better option.
> 
> While this PR also adds support for probabilities, #47410 is different because it does not use argmax, allowing for multilabel classification, as discussed in [#39173 (comment)](https://github.com/tensorflow/tensorflow/issues/39173#issuecomment-655022504)

This PR doesn't change original interface at all. It compares the shapes of `y_true` and `y_pred` to decide whether using argmax on `y_pred`.

File: tensorflow/python/keras/metrics_test.py [hunk 1]
```diff
@@ -1380,6 +1380,22 @@ def test_zero_and_non_zero_entries(self):
...
 
+  def test_y_pred_as_logits_or_probability_tensor(self):
+    y_true = [2, 0, 1, 1]
+    y_pred = [[2.5, 0.2, 2.7], [-1.4, 0.7, 1.2], [2., 3., 1.4], [2.2, 2.4, 1.5]]
+    
+    m_obj = metrics.MeanIoU(num_classes=3)
+    self.evaluate(variables.variables_initializer(m_obj.variables))
+    result = m_obj(y_true, y_pred)
+
+    # cm = [[0., 0., 1.],
```
Comment (qlzh727) [inline tensorflow/python/keras/metrics_test.py:1391]: Please include this section in the docstring as well, so that math of the example is more understandable.

**Review by qlzh727 — COMMENTED** [review]
Looks good. Thanks for the PR.

File: tensorflow/python/keras/metrics_test.py (hunk 1 above)
Comment (motionlife) [reply tensorflow/python/keras/metrics_test.py:1391]: done, thanks

**Review by deeb02 — APPROVED** [review]
LGTM. Thank you for the contribution.

**Comment by gbaned** [comment]
@motionlife Can you please address Ubuntu Sanity errors? Thanks!

**Comment by motionlife** [comment]
> @motionlife Can you please address Ubuntu Sanity errors? Thanks!

fixed, just found that K is no longer an alias of backend, replace K with backend, now All checks have passed

**Review by deeb02 — APPROVED** [review]
LGTM. Thanks again.

**Comment by gbaned** [comment]
@motionlife Can you please address Ubuntu Sanity errors? Thanks!

**Comment by motionlife** [comment]
> ess Ubuntu Sanity errors? Thanks!

The Ubuntu Sanity error log shows one line is too long, I break it into multiple lines, hope it will pass

**Comment by bhack** [comment]
Can we link with https://github.com/tensorflow/tensorflow/issues/39173 ?

**Review by deeb02 — APPROVED** [review]
LGTM.
